.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: Build macOS, Ubuntu, Windows
+
+on:
+  push:
```

Comment:
Let's build on `pull_request` events, not tag pushes-- this way we'll see build failures much sooner

Reply:
I am keeping this for testing via Actions tab. Will respond to this convo once changed to pull req events.

Reply:
Changed.

Reply:
in the event we do move towards GHA, it might be useful to build just on `push:` (like libsbp rust.yaml) but check if it has a tag to release so we dont have to build it twice and just upload the ones we built (don't think this affects performance tho since they run simultaneously anyways just reusing code i guess)

Reply:
@adrian-kong addressed below

Reply:
IMO we should keep this simple for now, we can optimize double building later

Reply:
> looking good so far, think we just need to port the commands from jenkins to run these:
> 
> ```
> cargo make lint
> cargo fmt -- --check
> 
> cargo make test
> ```

Addressed offline. Saving for another ticket.

Reply:
Can we move this back to triggering on `pull_requests`?

Reply:
From my previous response: 

I left a note # TEMPORARY change used to test achubbic because I wasn't sure how to run action workflows without tag pushing. This will be removed once ticket is in approval state with tests passing.

Alternatively, how can I run a actions workflow for a given commit using on: pull-request?

Reply:
It triggers when a pull request is opened or updated

> Runs your workflow when activity on a pull request in the workflow's repository occurs. For example, if no activity types are specified, the workflow runs when a pull request is opened or reopened or when the head branch of the pull request is updated.

See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

So all that should be needed is to push an update to the PR to run the new ci.yaml -- you'd like to run the workflow without changes you can force the workflow to re-run through the GitHub web UI or you can simply push an empty commit:

```
git commit --allow-empty -m "trigger ci"
git push
```

Reply:
> It triggers when a pull request is opened or _UPDATED_

Ah excellent. Thank you both.

Reply:
Change in question has been reverted.